tests/tensor_search/test_api.py

```diff
@@ -528,6 +529,23 @@ def test_invalid_structured_index_field_features(self):
                 self.assertIn("allFields", response.text)
                 self.assertIn("features", response.text)
 
+    def test_healthz_happy_pass(self):
+        response = self.client.get("/healthz")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {"status": "ok"})
+
+    def test_healthz_fails_if_exception_raised(self):
+        for cuda_exception in [
+            CudaDeviceNotAvailableError('Cuda device becomes unavailable'),
+            CudaOutOfMemoryError('Cuda device cuda:0 is out of memory')
+        ]:
+            with self.subTest(cuda_exception):
+                with patch("marqo.core.inference.device_manager.DeviceManager.cuda_device_health_check",
+                           side_effect=cuda_exception):
+                    response = self.client.get("/healthz")
+                    self.assertEqual(response.status_code, 500)
+                    self.assertIn(cuda_exception.message, response.json()['message'])
```

Comment:
We should have an integration test for this as well, on a marqo container

Reply:
Good point. You mean an API test, right? probably not using pymarqo since this is an internal endpont. Will add one once this is merged

Reply:
yep sorry I meant an api test.